src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/hooks/useOptimisticUpdate.ts

```diff
@@ -0,0 +1,13 @@
+import { useCallback, useEffect, useState } from 'react';
+
+export const useOptimisticUpdate = <T>(state: T) => {
+    const [value, setValue] = useState(state);
+
+    const rollback = useCallback(() => setValue(state), [state]);
```

Comment:
It is intentional. This hook allows to temporary change the state, before new reference arrives.

Reply:
Right, but should `rollback` use the previous `value` instead of the initial `state` from first render? Example:

```
it('should rollback to previous value', () => {
    const { result } = renderHook(state => useOptimisticUpdate(state), {
        initialProps: 1,
    });

    expect(result.current[0]).toEqual(1);
    act(() => result.current[1](2));
    expect(result.current[0]).toEqual(2);
    act(() => result.current[1](3));
    expect(result.current[0]).toEqual(3);
    act(() => result.current[2]());
    expect(result.current[0]).toEqual(2); // Fails
});
```

Reply:
It "rolls back" to `state`, because that is the external source of truth. In other words: "you can modify this internal state as much as you need, but when new external data arrives, reset it"